File: apps/backend/app/api/v1/health.py
"""Process liveness ve eklenti tipi dependency readiness uçları."""

from fastapi import APIRouter, Request

from app.api.v1.responses import HEALTH_LIVE, HEALTH_READY
from app.core.errors import ServiceNotReadyError
from app.schemas.health import LivenessResponse, ReadinessCheckResponse, ReadinessResponse
from app.services.health import ReadinessCheck, run_readiness_checks

router = APIRouter(prefix="/health", tags=["health"])


@router.get("/live", response_model=LivenessResponse, responses=HEALTH_LIVE)
async def live() -> LivenessResponse:
    """Process çalışıyorsa başarılıdır; dış servis kontrol etmez."""
    return LivenessResponse()


@router.get("/ready", response_model=ReadinessResponse, responses=HEALTH_READY)
async def ready(request: Request) -> ReadinessResponse:
    """Kayıtlı zorunlu bağımlılıkların trafik almaya hazır olduğunu doğrular."""
    checks: tuple[ReadinessCheck, ...] = request.app.state.readiness_checks
    if not checks:
        raise ServiceNotReadyError("Servis trafiğe hazır değil.")

    results = await run_readiness_checks(checks)
    if any(not result.ready for result in results):
        raise ServiceNotReadyError("Servis trafiğe hazır değil.")
    return ReadinessResponse(
        checks=[ReadinessCheckResponse(name=result.name) for result in results]
    )

File: apps/web/src/lib/api/schemas/health.ts
import * as z from "zod";

/** Process liveness cevabı; dış bağımlılık kontrol etmez. */
export const livenessResponseSchema = z.object({
  status: z.literal("ok"),
});

export type LivenessResponse = z.infer<typeof livenessResponseSchema>;

/** Readiness'e kayıtlı tek bir zorunlu bağımlılığın sonucu. */
export const readinessCheckResponseSchema = z.object({
  name: z.string(),
  status: z.literal("ok"),
});

export type ReadinessCheckResponse = z.infer<typeof readinessCheckResponseSchema>;

/** Trafik kabul etmeye hazır uygulama cevabı. */
export const readinessResponseSchema = z.object({
  status: z.literal("ready"),
  checks: z.array(readinessCheckResponseSchema).min(1),
});

export type ReadinessResponse = z.infer<typeof readinessResponseSchema>;
